Automattic/woocommerce-payments
# Send bank statement descriptors to payment intents

Fixed #4329

#### Changes proposed in this Pull Request








The usage of the statement descriptor in transactions is being addressed in this PR. This is the final part of the #3625's work.

For some portion of the transactions, the regular statement descriptor is being added to the payment intents already. Here we're making sure the descriptor is passed to the payment intent and the shortened descriptor is being included to take place on card payments, if enabled.

It also includes a small fix for the Blocks checkout when using UPE, where the `wcpay_selected_upe_payment_type` was not being sent.

#### Testing instructions





Test the descriptors set are sent to the payment intent.

![image](https://user-images.githubusercontent.com/10233985/149672714-eacccfb3-5d0d-4397-bad9-d9b9adb817f2.png)

There's a vast amount of combinations for this test, please try to test as much as you can:
- UPE enabled and disabled.
- Classic checkout and Blocks checkout.
- Card, non-card payment method and express checkout.
- `Add customer order number to the bank statement` enabled and disabled.

Also consider the following:
1. When no `Full bank statement` and `Shortened customer bank statement` are provided, the payment intent should either contain no statement descriptor or have the `Full bank statement` descriptor set.
2. When `Full bank statement` is provided and `Add customer order number to the bank statement` is disabled, the payment intent should contain the `Full bank statement`'s value in the statement descriptor regardless the payment method chosen.
3. In card transactions (express checkout included), when `Add customer order number to the bank statement` is enabled and `Shortened customer bank statement` is provided, the payment intent should contain the `Shortened customer bank statement`'s value concatenated with the order number in the statement descriptor.

**The test:**
1. Set values for both `Full bank statement` and `Shortened customer bank statement`.
2. Go to the [shop page](http://localhost:8082/shop/) and add a product to the cart.
3. Go to checkout and complete the order using the scenarios described above.
4. Go to [Payments](https://dashboard.stripe.com/test/payments) in the Stripe dashboard, find the payment you've just made and click on it.
5. Check whether the statement descriptor value (under Payment details section) match the expected value according to the scenario being tested.

-------------------


**Post merge**
